hack/run-operator-catalog.sh

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env bash
+
+# hack/run-operator-catalog.sh
+#
+# Run the operator from a catalog image.
+# Required environment variables:
+#
+# - CATALOG_IMG: catalog image to deploy
+# - CSV_VERSION: version tag of the cluster service version
+#
+# Optional environment variables
+#
+# - KUSTOMIZE_BIN: path to kustomize
+# - KUBECTL_BIN: path to kubectl (or equivalent command line)
+# - SED_BIN: path to GNU sed
+# - CATALOG_NAMESPACE: Namespace to deploy the catalog. Defaults to shipwright-operator.
+# - SUBSCRIPTION_NAMESPACE: Namespace to install the operator via an OLM subscription. Defaults to
+#   shipwright-operator.
+# - NAME_PREFIX: prefix to use for all resource names. Defaults to "shipwright-"
+
+set -e
+
+kustomize="${KUSTOMIZE_BIN:-${PWD}/bin/kustomize}"
+k8s="${KUBECTL_BIN:-kubectl}"
+sed="${SED_BIN:-sed}"
+catalogNamespace="${CATALOG_NAMESPACE:-shipwright-operator}"
+subNamespace="${SUBSCRIPTION_NAMESPACE:-shipwright-operator}"
+namePrefix="${NAME_PREFIX:-shipwright-}"
+
+if [[ -z ${CATALOG_IMG} ]]; then
+    echo "CATALOG_IMG environment variable must be set"
+    exit 1
+fi
+
+if [[ -z ${CSV_VERSION} ]]; then
+    echo "CSV_VERSION environment variable must be set"
+    exit 1
+fi
+
+echo "Adding replacements not supported by kustomize"
+${sed} -i -E "s|image: (.+)$|image: ${CATALOG_IMG}|g" config/catalog/catalog_source.yaml
+${sed} -i -E "s|startingCSV: (.+)$|startingCSV: shipwright-operator.v${CSV_VERSION}|g" config/subscription/subscription.yaml
+${sed} -i -E "s|sourceNamespace: (.+)$|sourceNamespace: ${catalogNamespace}|g" config/subscription/subscription.yaml
+${sed} -i -E "s|source: (.+)$|source: ${namePrefix}operator|g" config/subscription/subscription.yaml
+
+echo "Applying catalog source and subscription from kustomize"
+
+pushd config/catalog
+${kustomize} edit set namespace "${catalogNamespace}"
+${kustomize} edit set nameprefix "${namePrefix}"
+popd
+
+pushd config/subscription
+${kustomize} edit set namespace "${subNamespace}"
+${kustomize} edit set nameprefix "${namePrefix}"
+popd
+
+echo "Deploying catalog source"
+${kustomize} build config/catalog | ${k8s} apply -f -
+
+echo "Waiting for catalog source to be ready"
+attempts=1
+while [[ ${attempts} -le 5 ]]; do
+    echo "Checking the status of the catalog source - attempt ${attempts}"
+    if ${k8s} wait --for=condition=Ready pod -l "olm.catalogSource=${namePrefix}operator" -n "${catalogNamespace}"; then
+        attempts=100
+    fi
+    attempts=$(( attempts + 1 ))
+    sleep 10
+done
+
+if [[ ${attempts} -le 100 ]]; then
+    echo "Timed out waiting for the catalog source to be ready"
+    exit 1
+fi
+
+echo "Deploying subscription"
+${kustomize} build config/subscription | ${k8s} apply -f -
+
+echo "Waiting for operator deployment"
+
+attempts=1
+while [[ ${attempts} -le 10 ]]; do
+    echo "Checking the status of the operator rollout - attempt ${attempts}"
+    if ${k8s} rollout status deployment "${namePrefix}operator" -n "${subNamespace}"; then
```

Comment:
Here you can use the option `--timeout` and let Kubernetes client handle the retries for you. We do [the same on the CLI](https://github.com/shipwright-io/cli/blob/e36a2eae6d676b2401dff9d8b8449933d2aef75f/hack/install-tekton.sh#L14), please consider.

With the timeout option in place, we may also simplify the script not having to handle the attempt for-loop.

Reply:
I decided to use a single function to wait on pod status.